### PR TITLE
HW02 by Vadim Volovenko

### PR DESCRIPTION
**Pull request by Volovenko**

### CODE REVIEW

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+logs = File.readlines('data_for_t_01.txt', chomp: true)
+
+def task_1(logs)
+  empty_line = ''
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: We don't need this variable, just use `''` explicitly

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  error_message = logs.select { |log| log.downcase.include?('error') }
+  puts error_message.empty? ? empty_line : error_message
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: We usually don't use `puts` inside the method, a method just should return the value 
```ruby
puts task_1(logs)
```

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+logs = File.readlines('data_for_t_01.txt', chomp: true)
+
+def task_1(logs)
+  error_message = logs.select { |log| log.downcase.include?('error') }
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: `Method task_1 should return full text of the FIRST line with an error `. Try to use `.find`

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb (hunk 2 above)
Comment (Volovenko) [reply HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: done

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb (hunk 1 above)
Comment (Volovenko) [reply HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: done

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb (hunk 3 above)
Comment (Volovenko) [reply HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: done

File: HW02/Vadim_Volovenko/data_for_t_03.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/data_for_t_03.txt]: `txt` files should be placed in a separated folder together, for example `fixtures`

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+logs = File.readlines('data_for_t_01.txt', chomp: true)
+
+def task_1(logs)
+  error_message = logs.find { |log| log.downcase.include?('error') }
+  error_message.empty? ? '' : error_message
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: You can try to use the shortest version
```suggestion
  logs.find { |log| log.downcase.include?('error') }.to_s
```

File: HW02/Vadim_Volovenko/data_for_t_03.txt (hunk 4 above)
Comment (Volovenko) [reply HW02/Vadim_Volovenko/data_for_t_03.txt]: forgot about it, sorry. Done

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+logs = File.readlines('fixtures/data_for_t_01.txt', chomp: true)
+
+def task_1(logs)
+  logs.find { |log| log.downcase.include?('error') }.to_s
+end
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_01.rb]: 👍

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_02.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+DATE_REGEX = /[\[].*[\]]/.freeze
+IP_REGEX = /[\d.]{11}/.freeze
+PATH_REGEX = /POST\s\S+/.freeze
+
+logs = File.readlines('logs.txt', chomp: true)
+
+def task_2(logs)
+  empty_line = ''
+  logs_output = logs.select { |line| line =~ /POST/ }.map do |line|
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_02.rb]: This check is too weak, we don't have any line format checks here

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_03.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+TIME_REGEXP = /^\d{4}\-\d{2}\-\d{2}\s\d{2}\:\d{2}\:\d{2}.\d{1}/.freeze
+
+def logs
+  File.readlines('fixtures/data_for_t_03.txt', chomp: true).select do |log|
+    log =~ /Calling core/
+  end
+end
+
+def task_3(logs)
+  datetimestamps = logs.map do |log|
```
Comment (vladhilko) [inline HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_03.rb]: What if logs are empty?

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_02.rb (hunk 7 above)
Comment (Volovenko) [reply HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_02.rb]: I understand, fixed that

File: HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_03.rb (hunk 8 above)
Comment (Volovenko) [reply HW02/Vadim_Volovenko/vadim_volovenko_hw_02_t_03.rb]: added check for empty logs

**Pull request merged**